lacework/resource_lacework_integration_gcp_agentless_scanning.go

```diff
@@ -0,0 +1,384 @@
+								return !d.HasChanges(
+									"name", "resource_level", "resource_id", "org_level", "enabled",
+									"credentials.0.client_id",
+									"credentials.0.client_email",
+								)
+							},
+						},
+						"client_email": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"private_key": {
+							Type:      schema.TypeString,
+							Required:  true,
+							Sensitive: true,
+							DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+								// @afiune we can't compare this element since our API, for security reasons,
+								// does NOT return the private key configured in the Lacework server. So if
+								// any other element changed from the credentials then we trigger a diff
+								return !d.HasChanges(
+									"name", "resource_level", "resource_id", "org_level", "enabled",
+									"credentials.0.client_id",
+									"credentials.0.client_email",
+								)
+							},
+						},
+						"token_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "https://oauth2.googleapis.com/token",
+						},
+					},
+				},
+			},
+			"resource_level": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  api.GcpProjectIntegration.String(),
+				StateFunc: func(val interface{}) string {
+					return strings.ToUpper(val.(string))
+				},
+				ValidateFunc: func(value interface{}, key string) ([]string, []error) {
+					switch strings.ToUpper(value.(string)) {
+					case api.GcpProjectIntegration.String(),
+						api.GcpOrganizationIntegration.String():
+						return nil, nil
+					default:
+						return nil, []error{
+							fmt.Errorf("%s: can only be either '%s' or '%s'",
+								key,
+								api.GcpProjectIntegration.String(),
+								api.GcpOrganizationIntegration.String()),
+						}
+					}
+				},
+			},
+			"resource_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"created_or_updated_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_or_updated_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"org_level": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"bucket_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"scanning_project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"scan_frequency": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  24,
+			},
+			"scan_containers": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"scan_host_vulnerabilities": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"query_text": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"filter_list": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
```

Comment:
Would it be better to have filter_list be a list type?

Reply:
The `filter_list` is similar to the `monitored_accounts` parameter (https://github.com/lacework/terraform-provider-lacework/blob/main/lacework/resource_lacework_integration_aws_org_agentless_scanning.go#L73-L83) in the AWS org-level integration, so it's expected to be a comma separated list of projects/folders.

That said, we mainly did a string in AWS because CloudFormation is really limited in how a user would enter a list of items.  We could probably change it for GCP (would require some platform changes), but we may want to keep it as a string just for consistency.

Reply:
What I did with AWS was have the module user input a list: https://github.com/lacework/terraform-aws-agentless-scanning/blob/main/variables.tf#L136

And accept a list of strings for the module within the provider: https://github.com/lacework/terraform-provider-lacework/blob/07b1fa8fc26cce60b167fd521329514c85320510/lacework/resource_lacework_integration_aws_org_agentless_scanning.go#L73

Then join this into a string to send to the Lacework API: https://github.com/lacework/terraform-provider-lacework/blob/07b1fa8fc26cce60b167fd521329514c85320510/lacework/resource_lacework_integration_aws_org_agentless_scanning.go#L160

I recommend following the same process here, so go with @dmurray-lacework's suggestion and make `filter_list` into a `schema.TypeList`:
```go
	"filter_list": {
		Type:     schema.TypeList,
		Required: true,
		Elem: &schema.Schema{
			Type: schema.TypeString,
			StateFunc: func(val interface{}) string {
				return strings.TrimSpace(val.(string))
			},
		},
		Description: "",
	},
```

Reply:
Agreed. Makes sense. I changed this to `schema.TypeList` for gcp agentless scanning.

Reply:
Another note - I do not see us setting the `monitored_list` on reading the integration - https://github.com/lacework/terraform-provider-lacework/blob/07b1fa8fc26cce60b167fd521329514c85320510/lacework/resource_lacework_integration_aws_org_agentless_scanning.go#L221

I have added this for GCP. Should we do the same for AWS too?

Reply:
Ah, yes please